core/src/main/java/org/infinispan/util/concurrent/BlockingTaskAwareExecutorServiceImpl.java

```diff
@@ -43,26 +47,9 @@ public final void execute(BlockingRunnable runnable) {
       if (shutdown) {
          throw new RejectedExecutionException("Executor Service is already shutdown");
       }
-      if (runnable.isReady()) {
-         doExecute(runnable);
-      } else {
-         blockedTasks.offer(runnable);
-         //case: T1 is adding a task and T2 is releasing one. problem to solve:
-         //T1: is adding a new task. runnable.isReady() returns false
-         //T2: meanwhile, T2 releases a resources that is blocking the T1's runnable
-         //T2: also, T2 invokes checkForReadyTasks(), that does nothing because the queue is empty
-         //T1: continues and add the runnable to the queue
-         //problem: if no more interaction with this object is done, the runnable will be kept in the queue forever.
-
-         boolean checkPendingTasks;
-         synchronized (blockedTasks) {
-            //to synchronize with the thread invoking checkForReadyTasks();
-            checkPendingTasks = runnable.isReady();
-         }
-         if (checkPendingTasks) {
-            checkForReadyTasks();
-         }
-      }
+      //we no longer submit directly to the executor service.
+      blockedTasks.offer(runnable);
```

Comment:
with the default configuration no, we don't. but a user could inject their own executor service and we can end up with OOB threads executing the commands. 

IMO, most of the cases our executor service is used, so it may be good to go directly through the executor service if the task is ready.